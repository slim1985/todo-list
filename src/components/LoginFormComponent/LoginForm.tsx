export interface LoginFormProps {
    authenticate: () => void;
}

export function LoginForm({ authenticate }: LoginFormProps): JSX.Element {
    return (
        <div>
            <h1 className="text-center text-2xl retina:text-7xl mt-7 retina:mt-16 text-gray-600">
                Please Sign-In
            </h1>
            <div className="flex w-full justify-center mt-14 retina:mt-28">
                <button
                    className="border-solid border-gray-400 border-2 retina:border-4 py-2 retina:py-6 px-2 retina:px-6 rounded-lg retina:rounded-3xl retina:text-7xl"
                    onClick={authenticate}
                >
                    Sign-in by Google
                </button>
            </div>
        </div>
    );
}
