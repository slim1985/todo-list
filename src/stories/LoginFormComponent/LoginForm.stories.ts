import type { Meta, StoryObj } from '@storybook/react';
import { LoginForm } from '../../components/LoginFormComponent/LoginForm';

type Story = StoryObj<typeof meta>;

const meta: Meta<typeof LoginForm> = {
    component: LoginForm,
    parameters: {
        viewport: {
            defaultViewport: 'mobile1',
        },
    },
};

export const LoginFormSimple: Story = {
    name: 'Simple',
};

export default meta;
