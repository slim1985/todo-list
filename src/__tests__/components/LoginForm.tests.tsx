import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { LoginForm } from '../../components/LoginFormComponent/LoginForm';

const authenticate = jest.fn();

describe('LoginForm', () => {
    it('LoginForm renders correctly', () => {
        // Arrange.
        // Act.
        const { asFragment } = render(
            <LoginForm authenticate={authenticate} />,
        );

        // Assert.
        expect(asFragment()).toMatchSnapshot();
    });

    it('LoginForm authenticate event is called', () => {
        // Arrange.
        // Act.
        render(<LoginForm authenticate={authenticate} />);
        screen.getByText('Sign-in by Google').click();

        // Assert.
        expect(authenticate).toHaveBeenCalledTimes(1);
    });
});
