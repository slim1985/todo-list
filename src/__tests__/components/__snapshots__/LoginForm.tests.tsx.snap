// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm LoginForm renders correctly 1`] = `
<DocumentFragment>
  <div>
    <h1
      class="text-center text-2xl retina:text-7xl mt-7 retina:mt-16 text-gray-600"
    >
      Please Sign-In
    </h1>
    <div
      class="flex w-full justify-center mt-14 retina:mt-28"
    >
      <button
        class="border-solid border-gray-400 border-2 retina:border-4 py-2 retina:py-6 px-2 retina:px-6 rounded-lg retina:rounded-3xl retina:text-7xl"
      >
        Sign-in by Google
      </button>
    </div>
  </div>
</DocumentFragment>
`;
